### PR TITLE
refactoring & implementation of simple fingerprinter 

### PR DESCRIPTION
**Pull request by costezki**

### CODE REVIEW

**Review by cosletmihai — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -69,21 +69,19 @@ fuseki-create-test-dbs:
...
-	@ sudo rm -rf $(DATA_FOLDER)
-
-start-service: start-fuseki fuseki-create-test-dbs
-
-stop-service: stop-fuseki clean-data
-
+#clean-data:
+#	@ echo "$(BUILD_PRINT)Deleting the $(DATA_FOLDER)"
+#	@ sudo rm -rf $(DATA_FOLDER)
+#start-service: start-fuseki fuseki-create-test-dbs
```
[the remark this answers is not in the document]
Comment (costezki) [reply Makefile:80]: gone  in the next pr

File: fingerprint/service_layer/handlers.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    """
+        Calculate the fingerprint of a given endpoint and write the report in the output location.
+        Optionally the fingerprint calculation could be restricted to a particular named graph.
+    :param graph: a valid URI
+    :param sparql_endpoint_url:
+    :param output_location:
+    :return:
+    """
+    temp_dir = Path(output_location)
+    temp_dir.mkdir(parents=True, exist_ok=True)
```
[the remark this answers is not in the document]
Comment (costezki) [reply fingerprint/service_layer/handlers.py:39]: fixed

File: fingerprint/service_layer/handlers.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    with pkg_resources.path(fingerprint_report_templates, "fingerprint_report") as template_location:
+        copytree(template_location, temp_dir, dirs_exist_ok=True)
```
[the remark this answers is not in the document]
Comment (costezki) [reply fingerprint/service_layer/handlers.py:41]: fixed!
after extending eds4jinja2 funxtionality. 
I was lazy do to that in the first place, but now thanks to you, I did it.

File: fingerprint/service_layer/handlers.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    """
+        Read the default config json from the fingerprint_report and set the endpoint and the graph uri if necessary
+    :param sparql_endpoint_url:
+    :param graph:
+    :return: the new configuration
+    """
+    with pkg_resources.path(fingerprint_report_templates, "fingerprint_report") as resource_path:
+        config_dict = json.loads((resource_path / "config.json").read_bytes())
+        config_dict["conf"]["default_endpoint"] = sparql_endpoint_url
+        config_dict["conf"]["default_graph"] = graph
```
[the remark this answers is not in the document]
Comment (costezki) [reply fingerprint/service_layer/handlers.py:73]: that is to be checked internally. Currently not supported by the template

File: tests/unit/test_service_layer_handlers.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+def test_Accessing_templates():
+    with pkg_resources.path(fingerprint_report_templates, "fingerprint_report") as resource_path:
+        assert "fingerprint_report" in str(resource_path)
+
+
+def test_reading_template_confing_json_file():
+    with pkg_resources.path(fingerprint_report_templates, "fingerprint_report") as resource_path:
+        config_dict = json.loads((resource_path / "config.json").read_bytes())
+        assert "conf" in config_dict
+        assert "template" in config_dict
```
[the remark this answers is not in the document]
Comment (costezki) [reply tests/unit/test_service_layer_handlers.py:36]: these are 2 most stable keys in the ReportBuilder configs.

**Pull request merged**